expression/column.go

```diff
@@ -238,12 +243,23 @@ func (col *Column) EvalString(row types.Row, sc *variable.StatementContext) (str
 			return "", true, nil
 		}
 		res, err := val.ToString()
+		resLen := len([]rune(res))
+		if resLen < col.RetType.Flen && sc.PadCharToFullLength {
+			res = res + strings.Repeat(" ", col.RetType.Flen-resLen)
+		}
 		return res, err != nil, errors.Trace(err)
 	}
 	if row.IsNull(col.Index) {
 		return "", true, nil
 	}
-	return row.GetString(col.Index), false, nil
+	val := row.GetString(col.Index)
+	if sc.PadCharToFullLength {
+		valLen := len([]rune(val))
+		if valLen < col.RetType.Flen && sc.PadCharToFullLength {
```

Comment:
`sc.PadCharToFullLength` is already checked.

Reply:
@coocood  I think `sc.PadCharToFullLength` should be checked for both branch of `if col.GetType().Hybrid()` condition